internal/controller/pipelinerun_controller.go

```diff
@@ -41,15 +43,83 @@ var (
 
 // PipelineRunReconciler reconciles a PipelineRun object
 type PipelineRunReconciler struct {
-	Client client.Client
-	Scheme *runtime.Scheme
-	Config *config.ControllerConfig
+	Client     client.Client
+	Scheme     *runtime.Scheme
+	Config     *config.ControllerConfig
+	KiteClient *kite.Client
 }
 
 func (r *PipelineRunReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrllog.FromContext(ctx).WithName("PipelineRunController")
+	ctx = ctrllog.IntoContext(ctx, log)
+
+	// Get the PipelineRun object
+	var pipelineRun tektonv1.PipelineRun
+	if err := r.Client.Get(ctx, req.NamespacedName, &pipelineRun); err != nil {
+		// PipelineRun was deleted, nothing to do
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Only process completed PipelineRuns
+	if !pipelineRun.IsDone() {
```

Comment:
Instead of checking the PLR status here, could we add a callback to the checks in `SetupWithManager`?